ionoscloud/resource_server.go

```diff
@@ -948,6 +982,40 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	}
 
+	// Labels logic
+	if d.HasChanges("label") {
+		oldLabelsData, newLabelsData := d.GetChange("label")
+
+		// Delete the old labels.
+		oldLabels := getLabels(oldLabelsData)
+		for _, oldLabel := range oldLabels {
+			oldLabelKey := oldLabel["key"]
+			apiResponse, err := client.LabelsApi.DatacentersServersLabelsDelete(ctx, dcId, d.Id(), oldLabelKey).Execute()
+			logApiRequestTime(apiResponse)
+			if err != nil {
+				if httpNotFound(apiResponse) {
+					log.Printf("[WARNING] label with key %s has been already removed from server %s\n", oldLabelKey, d.Id())
+				} else {
+					return diag.FromErr(fmt.Errorf("[label update] an error occured while deleting label with key: %s, server ID: %s", oldLabelKey, d.Id()))
+				}
+			}
+		}
+
+		// Create the new labels.
+		newLabels := getLabels(newLabelsData)
+		for _, newLabel := range newLabels {
+			newLabelKey := newLabel["key"]
+			newLabelValue := newLabel["value"]
+			newLabelResource := ionoscloud.LabelResource{
+				Properties: &ionoscloud.LabelResourceProperties{Key: &newLabelKey, Value: &newLabelValue},
+			}
+			_, apiResponse, err := client.LabelsApi.DatacentersServersLabelsPost(ctx, dcId, d.Id()).Label(newLabelResource).Execute()
```

Comment:
this is boilerplate code, it looks the same for label creation in `resourceServerCreate`, I leave this as it is for now but I will think about a way to better organize this since we will also use the logic for other resources: volumes, datacenters, etc.

Reply:
@avirtopeanu-ionos done in 586238e4619fe7d41d00334bfc01c629d834a759